pkg/controller/epgcache.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2021 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"github.com/noironetworks/aci-containers/pkg/apicapi"
+	"github.com/noironetworks/aci-containers/pkg/metadata"
+
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+func (cont *AciController) clearFaultInstances() {
+	args := []string{
+		"order-by=vmmClusterFaultInfo.modTs|desc",
+	}
+	faultUri := fmt.Sprintf("/api/node/class/vmmClusterFaultInfo.json?%s", strings.Join(args, "&"))
+	apicresp, err := cont.apicConn.GetApicResponse(faultUri)
+	if err != nil {
+		return
+	}
+	subString := fmt.Sprintf("comp/prov-Kubernetes/ctrlr-[%s]-%s/injcont/info/", cont.config.AciVmmDomain, cont.config.AciVmmDomain)
+	for _, obj := range apicresp.Imdata {
+		for _, body := range obj {
+			dn, ok := body.Attributes["dn"].(string)
+			if ok {
+				cont.indexMutex.Lock()
+				if strings.Contains(dn, subString) {
+					cont.apicConn.DeleteDn(dn)
+					cont.log.Debug("Clearing existing Fault Instance: ", dn)
+				}
+				cont.indexMutex.Unlock()
+			}
+		}
+	}
+	return
+}
+
+func (cont *AciController) getSubString(dn string) string {
+	delimiterString := fmt.Sprintf("uni/vmmp-Kubernetes/dom-%s/eppd-[", cont.config.AciVmmDomain)
+	epgDn := strings.Split(dn, delimiterString)[1]
+	epgDn = strings.Replace(epgDn, "]", "", -1)
+	return epgDn
+}
+
+func (cont *AciController) epgClassChanged(obj apicapi.ApicObject) {
+	dn := obj.GetAttrStr("dn")
+	epgDn := cont.getSubString(dn)
+	cont.indexMutex.Lock()
+	if ok := !cont.contains(cont.cachedVRFDns, epgDn); ok {
```

Comment:
It's kind of strange that we're storing EPG DNs in 'cachedVRFDns'. Does it make sense to have a different var for this? Same comment applies to func names below